kategory-docs/docs/docs/integrations/kindedj/README.md

```diff
@@ -0,0 +1,82 @@
+---
+layout: docs
+title: KindedJ
+permalink: /docs/integrations/kindedj/
+---
+
+## KindedJ
+
+[KindedJ](https://github.com/KindedJ/KindedJ/) is a project to create an interop layer between multiple libraries that emulate [higher kinds]({{ '/docs/patterns/glossary' | relative_url }}) in the JVM
+using lightweight higher kind polymorphisms. You can read more about this emulation in [KindedJ's Readme](https://github.com/KindedJ/KindedJ/blob/master/README.md).
+
+### Generation of KindedJ interfaces
+
+Annotating a [datatype]({{ '/docs/patterns/glossary' | relative_url }}) with `@higherkind` will create a typealias `<datatype>KindedJ` that'll be directly available to be used by any other libraries that use [KindedJ](https://github.com/KindedJ/KindedJ/).
+Due to the difference in [generics between Kotlin and Java](https://kotlinlang.org/docs/reference/generics.html) this typealias will only be generated if all the generic parameters of the datatype are invariant.
+
+To work with datatypes whose generic parameters are not invariant requires a conversion layer.
+
+### Conversion module
+
+The kategory-kindedj module provides a conversion layer between Kategory and KindedJ where the kotlin generics `in` and `out` get on the way.
+
+It uses an integration construct called `Convert`, alongside several extension functions.
+
+#### Using KindedJ in Kategory from Kotlin
+
+Using the conversion layer we're capable of using an intermediate representation of any KindedJ datatype generically by converting it into a `HK2<ConvertHK, F, A>`,
+where F is the [original representation of the container]({{ '/docs/patterns/glossary' | relative_url }}).
+
+Let's see an example of a Java class defined by a third party:
+
+```java
+class IdJ<A> implements io.kindedj.Hk<IdJHK, A> {
+    public A a;
+}
+```
+
+You can convert it to Kategory using the extension function `fromKindedJ()`:
+
+```kotlin
+val idj = IdJ(1)
+
+val id: HK2<ConvertHK, IdJHK, A> = idj.fromKindedJ()
+```
+
+and convert it back using `convert()`:
+
+```kotlin
+val idj2: io.kindedj.Hk<IdJHK, A> = id.convert()
```

Comment:
This identifier is too broad in meaning and should be instead something like `toKindedJ()` that denotes the direction of the conversion with more clear semantics.